### DISCUSSION TITLE
Spatial coordinates based evaluation

### DISCUSSION THREAD

**Discussion by somsubhro88** [pyvista/pyvista#1797] [Q&A] [answered]
Hello everyone,

First of all, thank you for the effort in making this library. I have a specific question to my application.

I have a .vtu file which has a mesh and associated data. Below is an image of what the mesh looks like -
![image](https://user-images.githubusercontent.com/20040210/141499395-ec5f7523-4e56-4c64-9595-8c5cd0e039fb.png)

Below is an the same mesh with "psi" , the array in the mesh I would like to work with -
![image](https://user-images.githubusercontent.com/20040210/141499562-9d5b95ad-ceaa-4252-9efd-ada0ff76c7da.png)

What I would like to do is explained in the image below - I would like to extract a list of "psi" values along the y-axis and x-axis, There is a long version of solving this - extracting each value separately and then matching them up. I was wondering if there's clever way of doing this using already defined functions? My primary goal is to be able to measure the circular feature that you see in the images, and my criteria for measuring is "psi".
![image](https://user-images.githubusercontent.com/20040210/141500250-7f6e0128-f02b-49d9-9a2a-ecdbad360242.png)

I have tried with [sample_over_line](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSet.sample_over_line.html?highlight=sample%20over%20line#pyvista.DataSet.sample_over_line), but I am not able to extract the 'psi' value specifically.

Any help or guidance would be greatly appreciated! I have also attached the .vtu file below.
[solution-10000000.zip](https://github.com/pyvista/pyvista/files/7528990/solution-10000000.zip)
.

**Comment by MatthewFlamm** [comment, marked as answer]
In the example you linked to the data is copied over to the sampled lines.  What does this mean exactly 

> but I am not able to extract the 'psi' value specifically.

If you are trying to measure the circular feature, you could also try the contour filter using an isovalue.  It kind of sounds like you are trying to implement this already.

**Reply by somsubhro88** [reply]
Hello @MatthewFlamm. Thank you for your response.

When I use sample_over_line, it doesn't seem to be carrying any of the 'psi' array data. I am easily able to create a PolyData for the defined line (let's say from [0,10,0] to [20,10,0]), but this PolyData has no psi array data. Maybe I need to extract the psi data on a XY-plane, as in the example of the help guide on sample_over_line? I am not sure how to do this though.

**Reply by MatthewFlamm** [reply]
I'm still confused about this.  Can you give a minimal example, or at least a snippet of what you are doing?  Here is an example that shows that the data exists:

```py
import pyvista
mesh = pyvista.UniformGrid((10, 10, 10))
mesh["y_coord"] = mesh.points [:, 2]  # store y coordinate in point data
line_mesh = mesh.sample_over_line([0, 0, 0], [0, 5, 5])
line_mesh["y_coord"]  # data exists here
```

I also tried this with cell data, and achieved the same result.

**Reply by somsubhro88** [reply]
Hello @MatthewFlamm ,

For the following 2D data for "psi" that I'm interested in - 
![image](https://user-images.githubusercontent.com/20040210/143038179-5326cbdb-4154-426f-9412-6e688756c59a.png)

I'm using the following code:
```
#access points
x_coords = mesh.points[:,0]
y_coords = mesh.points[:,1]

line_mesh = mesh.sample_over_line([0.009, 0.01, 0], [0.011, 0.01, 0]) #this line can be seen in the plot above (the line representing a horizontal line through the top points)

#Your suggestion to access data from line_mesh
line_mesh["y_coords"] #Gave me an error: data array (x_coords) not present in this dataset:

#Following from the sample_over_line documentation example - Didn't work
pl = pv.Plotter()

_ = pl.add_mesh(y_coords, scalars='psi',line_width = 10)

pl.show()
```
I do not fully understand the sample_over_line works. Maybe the mesh type "UnstructuredGrid" is the reason for sample_over_line to not work?

**Reply by MatthewFlamm** [reply]
I would expect something like this:

```py
mesh = ... # read in mesh or create it
assert "psi" in mesh.array_names   # psi must be in mesh

# Triple check that this line is inside your dataset.
# I moved y-position to be somewhere central-ish.
# Make sure z=0 for the mesh.
line_mesh = mesh.sample_over_line([0.009, 0.0094, 0], [0.011, 0.0094, 0])

assert "psi" in line_mesh.array_names  # psi should be in line_mesh

pl = pv.Plotter()
pl.add_mesh(line_mesh, scalars="psi", line_width=10)
pl.show()
```

In my example above, it also works using an unstructured grid:

```py
mesh = pyvista.UniformGrid((10, 10, 10)).cast_to_unstructured_grid()
```

**Comment by RichardScottOZ** [comment]
Kdtree you can try? Or make your plane as a mesh and sample one onto the other?

**Reply by somsubhro88** [reply]
Hello @RichardScottOZ. Thanks for the idea. I'll try as well. For now I've managed to go the "Excel" route by converting the data set I wish to investigate into a Pandas DataFrame (containing the spatial coordinates of each node along with point_data) and performing operations on this DataFrame. Then I use alphashape to convert the points to a polygon which represents the semi-circular feature in the image above.

**Comment by bluetyson** [comment]
Looks ok to me :-

![image](https://user-images.githubusercontent.com/17399794/154821043-ff9fc036-83ef-4ca1-8de3-59065a3ba563.png)


![image](https://user-images.githubusercontent.com/17399794/154821011-4a751393-3bce-4ce4-b39c-4fc42d23fada.png)

![image](https://user-images.githubusercontent.com/17399794/154821016-e6572365-db15-43ac-b1ce-4c37682260a1.png)
